### PR TITLE
Update mapbox-gl

### PR DESCRIPTION
**Pull request by lsarni**
TODO:
- [ ] Build new version
- [ ] Publish new version

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vue2mapbox-gl",
-  "version": "0.13.1",
+  "version": "0.15.0",
```
Comment (lsarni) [inline package.json:3]: On npm it's version `0.14.0` already https://www.npmjs.com/package/vue2mapbox-gl

File: package.json [hunk 2]
```diff
@@ -48,11 +48,10 @@
     ]
   },
   "dependencies": {
-    "@mapbox/mapbox-gl-geocoder": "^4.4.1",
-    "resize-observer-polyfill": "^1.5.1"
```
Comment (lsarni) [inline package.json:52]: Browser compatibility has improved, we don't need the polyfill https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver#browser_compatibility

File: src/components/VMapbox.vue [hunk 3]
```diff
@@ -211,9 +211,10 @@ export default {
       this.addLayers()
     })
 
-    // If the style was changed, wait for the styledata to be loaded and re-add all the layers
+    // If the style was changed, wait for the style to be loaded and re-add all the layers
+    // https://github.com/mapbox/mapbox-gl-js/issues/4006
     this.$on('style:update', () => {
-      this.$once('mb-styledata', () => {
+      this.$once('mb-style.load', () => {
```
Comment (lsarni) [inline src/components/VMapbox.vue:217]: On both the doc and the example they use `style.load` for this https://github.com/mapbox/mapbox-gl-js/blob/937f67f321f5419c0e82e78e262553330e59c8bd/src/ui/events.js#L1502

File: src/components/v-mapbox-layer.js [hunk 4]
```diff
@@ -31,16 +31,9 @@ export default {
...
-    isInitialized: false
-  }),
-
   methods: {
     deferredMountedTo() {
-      if(!this.isInitialized) {
-        this.renderLayer();
-        this.isInitialized = true;
-      }
+      this.renderLayer();
```
Comment (lsarni) [inline src/components/v-mapbox-layer.js:36]: If the map style changed `isInitialized` was true, so the layers weren't being re-rendered.

File: src/components/VMapbox.vue [hunk 5]
```diff
@@ -178,7 +178,7 @@ export default {
     },
 
     resize() {
-      if (this.map) {
+      if (this.map && this.map.getCanvas()) {
```
Comment (lsarni) [inline src/components/VMapbox.vue:181]: To prevent this error 
image

File: src/stories/index.stories.js [hunk 6]
```diff
@@ -11,10 +11,12 @@ import Vue2MapboxGl from '../main.js'
 
 Vue.use(Vue2MapboxGl)
 
+const accessToken = process.env.MAPBOX_ACCESS_TOKEN;
+
 const mapTemplate = `
 <v-mapbox
  map-style="mapbox://styles/mapbox/satellite-streets-v10"
- access-token="pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4NXVycTA2emYycXBndHRqcmZ3N3gifQ.rJcFIG214AriISLbB6B5aw"
```
Comment (lsarni) [inline src/stories/index.stories.js:17]: This one isn't working anymore, but I didn't want to commit a new one

File: src/components/VMapbox.vue [hunk 7]
```diff
@@ -211,10 +219,11 @@ export default {
...
     })
 
-    // If the style was changed, wait for the styledata to be loaded and re-add all the layers
+    // If the style was changed, wait for the style to be loaded and re-add all the layers
+    // https://github.com/mapbox/mapbox-gl-js/issues/4006
     this.$on('style:update', () => {
-      this.$once('mb-styledata', () => {
-        this.addLayers()
+      this.$once('mb-style.load', () => {
+        this.refreshLayers()
```
Comment (lsarni) [inline src/components/VMapbox.vue:226]: Only the map layers get removed when the style changes, `addLayers` duplicated the controllers

**Comment by SiggyF** [comment]
Looks good, merged

**Pull request merged**